### ISSUE TITLE
Make documentation clearer on acceptable values for json property

### ISSUE THREAD

**Issue by atavakoli** [request/request#2450]
The documentation is currently not clear what values can be passed to JSON, and what the values of other fields should be based on JSON's value. According to the documentation for the `body` option:

> - `body` - entity body for PATCH, POST and PUT requests. Must be a `Buffer`, `String` or `ReadStream`. If `json` is `true`, then `body` must be a JSON-serializable object.

This implies that `json` is a boolean field; all the examples show `json` being populated with `true`. However, unit tests show that `json` can also be populated with an object. The documentation for `json` only says:

> - `json` - sets `body` to JSON representation of value and adds `Content-type: application/json` header.  Additionally, parses the response body as JSON.

But doesn't say what "sets" means exactly. It would be good to clarify exactly what's acceptable for this field, and what happens in the various cases.  For example,

> - `json` - **when set to `true`,** sets `body` to JSON representation of value and adds `Content-type: application/json` header.  Additionally, parses the response body as JSON. **When set to an object ...**

It's also not clear on what happens to `form` when `json` is used, either as a boolean or object. In general, it's not clear what the behavior is when `form` and `body` are specified together.

(Apologies in advance if this is already documented somewhere that I missed.)

**Comment by simov** [comment]
- form, body and json are used to set the request body, meaning you can't use all of them at once

- form sets the body to url encoded string

- json sets the body to json stringified string

- json true tries to json.parse the response body (+ some edge case when used with body)

Other than that this options is really messed up, but we can't fix it without major release. Also we have tests for each and every combination between these combinations.

**Comment by atavakoli** [comment]
It would be good to document include the above in the README; for example, the fact that only one of `form`, `body`, or `json` should be used in any given request (even if there's code for how to gracefully handle other cases) would be very useful to clearly document.

Also, every example in the README and the examples folder, and many (all?) test-cases use `json: true` with a body, so I'm not sure I'd consider it an edge case.

It would be great to fix the code, but even today having a clearer set of documentation on how these fields are intended to be used (i.e. your bullet-list above) would help a lot.

**Comment by simov** [comment]
We try to keep the options list as short and concise as possible. Also if you take a closer look at the option list you'll notice that several options are separated by horizontal line. All options grouped with body are used to set the request body in one way or another.

**Comment by Jokero** [comment]
I agree that documentation is not entirely clear. For example I didn't know before reading this issue that I can use `json()` method:

```js
rp.post('/test').json({ a: 1, b: 2 })
```

There are no any examples of it.
